media/public/include/IMediaPipeline.h

```diff
@@ -1241,23 +1241,33 @@ class IMediaPipeline
     virtual bool renderFrame() = 0;
 
     /**
-     * @brief Set level and transition of audio attenuation.
-     *        Sets the current volume for the pipeline (0.0 silent -> 1.0 full volume)
+     * @brief Set the target volume level and transition duration with easing type.
+     *        If duration is 0 milliseconds, then volume is immediately set
      *
-     * @param[in] volume : Target volume level (0.0 - 1.0)
+     * @param[in] targetVolume : Target volume level (0.0 - 1.0)
+     * @param[in] duration : (Optional) Duration of the volume transition in milliseconds
+     * @param[in] type : (Optional) Easing type for the volume transition
      *
-     * @retval true on success false otherwise
+     * @retval true on success.
      */
-    virtual bool setVolume(double volume) = 0;
+    virtual bool setVolume(double targetVolume, uint32_t duration = 0, EaseType type = EaseType::EASE_LINEAR) = 0;
 
     /**
-     * @brief Get current audio level. Fetches the current volume level for the pipeline.
+     * @brief Get current audio level or target volume if no fade transition. Fetches the current volume level for the pipeline.
```

Comment:
I think this description can stay as it is as it should just get whatever volume we are at now.